pkg/component/controller/cplb/cplb_reconciler.go

```diff
@@ -109,22 +123,154 @@ func (r *CPLBReconciler) maybeUpdateIPs(endpoint *corev1.Endpoints) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
 
-	// endpoints are not guaranteed to be sorted by IP address
-	slices.Sort(newAddresses)
+	for _, addr := range newAddresses {
+		if _, ok := r.healthCheckers[addr]; !ok {
+			r.healthCheckers[addr] = r.newHealthChecker(ctx, restConfig, addr, nil)
+		}
+	}
 
-	if !slices.Equal(r.addresses, newAddresses) {
-		r.addresses = newAddresses
-		r.log.Infof("Updated the list of IPs: %v", r.addresses)
-		select {
-		case r.updateCh <- struct{}{}:
-		default:
+	shouldNotify := false
+	for addr := range r.healthCheckers {
+		if !slices.Contains(newAddresses, addr) {
+			r.healthCheckers[addr].Stop()
+			delete(r.healthCheckers, addr)
+			shouldNotify = true
 		}
 	}
+	if shouldNotify {
+		r.updateCh <- struct{}{}
```

Comment:
Any reason not to keep the old way of doing it? Especially since we're holding the lock.


```suggestion
		select {
		case r.updateCh <- struct{}{}:
		default:
		}
```